include/zephyr/net/net_pkt.h

```diff
@@ -822,6 +838,35 @@ static inline void net_pkt_set_ipv6_fragment_id(struct net_pkt *pkt,
 }
 #endif /* CONFIG_NET_IPV6_FRAGMENT */
 
+#if defined(CONFIG_NET_IPV4_FRAGMENT) || defined(CONFIG_NET_IPV6_FRAGMENT)
```

Comment:
**A more general remark, definitely not something this PR should do;**

The code base has a lot of `#if defined(CONFIG_NET_IPV4_<feature>) || defined(CONFIG_NET_IPV6_<feature>)`,
I think we could use a hidden Kconfig to our advantage here:

```cfg
config NET_IP_FRAGMENT
	bool
	default y
	depends on NET_IPV4_FRAGMENT || NET_IPV6_FRAGMENT
```

And use this option in code:

```c
#ifdef CONFIG_NET_IP_FRAGMENT
/* Insert code */
#endif
```

Disclaimer: This is the lazy me, who wants to type/copy less, talking.

Reply:
Good point actualy, I've added one more commit introducing the hidden symbol for IP fragmentation.